cql3/expr/expression.cc

```diff
@@ -103,7 +103,13 @@ managed_bytes_opt get_value(const column_value& col, const column_value_eval_bag
         if (!col_type->is_map()) {
             throw exceptions::invalid_request_exception(format("subscripting non-map column {}", cdef->name_as_text()));
         }
-        const auto deserialized = cdef->type->deserialize(managed_bytes_view(*data.other_columns[data.sel.index_of(*cdef)]));
+        int32_t index = data.sel.index_of(*cdef);
+        if (index == -1) {
+            throw exceptions::invalid_request_exception(
```

Comment:
I have a new complaint (sorry). You throw here an "invalid request exception" which I assume (but please correct me if I'm wrong) will be translated into a CQL error saying there is something wrong with the request - which tells the user that it is pointless to retry the request because his request is wrong. However, it's not that the **CQL request** is wrong here - the request is probably fine but we have some internal error (like schema version mismatch). So wouldn't it be better to throw some sort of internal error?

Reply:
Hm, right, that one should be a runtime_error or perhaps a logic_error, that gets translated to an "internal server error" when going back to the client